qiskit/circuit/store.py

```diff
@@ -59,6 +59,9 @@ class Store(Instruction):
     :class:`~.circuit.Measure` is a primitive for quantum measurement), and is not safe for
     subclassing."""
 
+    # This is a compiler/backend intrinsic operation, separate to any quantum processing.
+    _directive = True
```

Comment:
I'm fine with this, but this basically means we're treating this as always supported through the transpiler regardless of what the target says. Is that the correct behavior for this? I'm mostly wondering for backends that don't have any classical logic support, and how we intend to indicate this to backends. For other control flow operations they have to be explicitly in the target to work. Is the answer just documentation?

Reply:
Short of adding more magic fields to `supported_instruction` `Target`, which could never really capture all the constraints of dynamic-circuits requirements already, or some sort of global flag on `Target`/`BackendV2` that we haven't really prepared to put in place, I think documentation is the answer.

Dynamic circuits have always been tricky through these tests, because there's _so much_ that's never really been possible to represent as a restriction to the transpiler (nesting can only be to `n` levels, max of `m` blocks, switches must have a `default`, etc etc).

Reply:
Let just go with documenting this for now as part of the 1.1.x final documentation, I'll convert this into an issue to track. Both as an upgrade note in the final release notes and also as a guidance in the writing a backend section of the docs.

I think for 1.2.x we can look at adding a global flag in the target for something like supports store, etc to provide a high level indicator for this kind of capability above the per instruction look ups. We'd probably need to do this into an opt-in way (so it defaults true) but I think having the capability to express this in the target is what we'll want longer term.